Common/Orders/Fills/ImmediateFillModel.cs

```diff
@@ -379,6 +383,59 @@ public OrderEvent MarketOnCloseFill(Security asset, MarketOnCloseOrder order)
             return fill;
         }
 
+        /// <summary>
+        /// Volume Weighted Average Price Fill Model. Return an order event with the fill details
+        /// </summary>
+        /// <param name="asset">Asset we're trading with this order</param>
+        /// <param name="order">Order to be filled</param>
+        /// <returns>Order fill information detailing the average price and quantity filled.</returns>
+        public OrderEvent VwapFill(Security asset, VwapOrder order)
+        {
+            decimal vwap;
+
+            lock (_vwapLocker)
```

Comment:
do we need this lock? There's exactly one fill model instance per security and they're invoked synchronously via `BacktestingBrokerage.Scan`

Reply:
I _think_ it is needed, `BacktestingBrokerage.Scan` is called from both `ProcessSynchronousEvents` and `ProcessAsynchronousEvents`

Reply:
Perhaps it's safest to leave it in, but `BacktestingBrokerage.Scan` is already wrapped in a lock so it's guaranteed to run synchronously and not re-enter -- that being said, we shouldn't make decisions in this model based on its consumers, so lets leave it :)